src/InstrumentationEngine/ProfilerManagerWrapper.cpp

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "stdafx.h"
+#include "ProfilerManager.h"
+#include "ProfilerManagerWrapper.h"
+
+MicrosoftInstrumentationEngine::CProfilerManagerWrapper::CProfilerManagerWrapper(GUID instrumentationMethodGuid, CProfilerManager* pProfilerManager)
+{
+    WCHAR wszCurrentMethodGuid[40] = { 0 };
+    if (0 != ::StringFromGUID2(instrumentationMethodGuid, wszCurrentMethodGuid, 40))
+    {
+        m_wszInstrumentationMethodGuid = wszCurrentMethodGuid;
+    }
+
+    m_pProfilerManager = pProfilerManager;
+}
+
+MicrosoftInstrumentationEngine::CProfilerManagerWrapper::~CProfilerManagerWrapper()
+{
+
+}
+
+// IProfilerManager Methods
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::SetupProfilingEnvironment(_In_reads_(numConfigPaths) BSTR rgConfigPaths[], _In_ UINT numConfigPaths)
+{
+    return m_pProfilerManager->SetupProfilingEnvironment(rgConfigPaths, numConfigPaths);
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::AddRawProfilerHook(_In_ IUnknown *pUnkProfilerCallback)
+{
+    return m_pProfilerManager->AddRawProfilerHook(pUnkProfilerCallback);
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::RemoveRawProfilerHook()
+{
+    return m_pProfilerManager->RemoveRawProfilerHook();
+}
+
+HRESULT MicrosoftInstrumentationEngine::CProfilerManagerWrapper::GetCorProfilerInfo(_Outptr_ IUnknown **ppCorProfiler)
+{
+    HRESULT hr = S_OK;
+    CComPtr<IUnknown> pCorProfiler;
```

Comment:
A lot of these methods don't really do anything... why not just forward the call directly to the underlying m_pProfilerManager and not worry about Attach/Detach, local variables, etc?

Reply:
I was not thinking clearly

---
In reply to: [331295260](https://github.com/microsoft/CLRInstrumentationEngine/pull/146#discussion_r331295260) [](ancestors = 331295260)